src/Auth0.php

```diff
@@ -298,6 +324,14 @@ public function exchange() {
       return false;
     }
 
+    $state = $this->getState();
```

Comment:
@glena I've done it this way to avoid any BC, if stateHandler has no state then it will skip, ideally I wanted to log this but didn't see any precedent of logging.
For example use case: User calls /authorize manually then redirects to Auth->exchange (which would then attempt SessionStateHandler validation but has no record of the state)

Reply:
🤔 if state does not match we should fail (otherwise there is not value in checking it). Maybe in that case provide a state param to exchange? or a configuration to disable the verification...

Reply:
Well this was for the case of not having a stored state as the user called `/authorize` themselves but then decided to use `exchange()` in their callback. Which expects to find it in the StateHandler.
I've moved this to be an education issue rather than trying to solve all scenarios in code.